include/avif/avif.h

```diff
@@ -48,6 +48,13 @@ extern "C" {
 #define AVIF_API
 #endif // defined(AVIF_DLL)
 
+#if defined(AVIF_ENABLE_NODISCARD) || (defined(__cplusplus) && __cplusplus >= 201700L) || \
+    (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L)
```

Comment:
why the || conditions? If we have an explicit flag to turn it on (AVIF_ENABLE_NODISCARD) it seems weird to also turn it on based on some other conditions

Reply:
I agree, I thought it would be a good thing to have it by default if possible. Maybe not.